### PR TITLE
vnc_updater: use [enter|leave]_critical_section replace sched_[un]lock

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
sched_[un]lock can not prohibit pre-emption in smp, https://github.com/apache/nuttx/issues/1137

### CODE REVIEW

File: drivers/video/vnc/vnc_updater.c [hunk 1]
```diff
@@ -181,23 +182,24 @@ static FAR struct vnc_fbupdate_s *
...
 
-  sched_lock();
   vnc_sem_debug(session, "Before alloc", 0);
+  flags = enter_critical_section();
 
   nxsem_wait_uninterruptible(&session->freesem);
 
   /* It is reserved.. go get it */
 
   update = (FAR struct vnc_fbupdate_s *)sq_remfirst(&session->updfree);
```
Comment (pkarashchenko) [inline drivers/video/vnc/vnc_updater.c:199]: Can `session->updfree` be accessed from different cores?

Comment (xiaoxiang781216) [reply drivers/video/vnc/vnc_updater.c:199]: Yes, since it's accessed by two thread and then possible on SMP.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by acassis** [comment]
@xiaoxiang781216 do you have a VNC client running on NuttX already? I'm willing to add one (at apps/ instead of kernel), I just took a look at https://github.com/wolfgangB33r/VNCView and it seems straightforward.

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 do you have a VNC client running on NuttX already?

No, we just run VNCView from PC to connect NuttX.

> I'm willing to add one (at apps/ instead of kernel), I just took a look at https://github.com/wolfgangB33r/VNCView and it seems straightforward.

Yes, it will be useful in the reverse direction.

**Pull request merged**